### PR TITLE
Fix nulls that are introduced during index partitioning

### PR DESCRIPTION
**Pull request by phofl**
cc @hendrikmakait 

Same as what we fixed in dask/dask before Christmas

ref #525

### CODE REVIEW

File: dask_expr/_shuffle.py
Comment (hendrikmakait) [inline dask_expr/_shuffle.py:250]: This condition feels overly complicated. If I'm not missing anything, this simplifies to `nset.issubset(set(frame.columns))`.

File: dask_expr/_merge.py
Comment (hendrikmakait) [inline dask_expr/_merge.py:287]: Why does this rely on `tasks`?

Comment (phofl) [reply dask_expr/_merge.py:287]: To keep the hashing consistent, not ideal but everything else loses data at the moment

Comment (phofl) [reply dask_expr/_merge.py]: Cleaned this up now after we removed the bug in dask/dask

File: dask_expr/_shuffle.py
Comment (phofl) [reply dask_expr/_shuffle.py]: Removed altogether similar to dask/dask

**Comment by phofl** [comment]
Turns out this gave me so much trouble because dask/dask had a bug, I am disabling the release ci until the next release because we will get a bunch of failures there. Won't cut a new dask-expr release until dask/dask releases. We will have to bump the min dask version then as well

File: .github/workflows/test.yaml
Comment (hendrikmakait) [inline .github/workflows/test.yaml:28]: Is this intended?

**Review by hendrikmakait — APPROVED** [review]
Generally LGTM; one question that needs to be clarified before merging, but there's no need to go through another review. Thanks, @phofl!

File: .github/workflows/test.yaml
Comment (phofl) [reply .github/workflows/test.yaml:28]: Yes

We are currently incompatible with released dask dask

Comment (hendrikmakait) [reply .github/workflows/test.yaml:28]: Makes sense, in an ideal world we should have split that into a separate PR.

Comment (phofl) [reply .github/workflows/test.yaml:28]: I actually merged this already in another PR before this one, not sure why the diff is still the way it is

**Pull request merged**
